From 790aad31291d199d8fecc84767ce59328d60cd5e Mon Sep 17 00:00:00 2001
From: "Song, Yoong Siang" <yoong.siang.song@intel.com>
Date: Tue, 5 Nov 2019 21:55:24 +0800
Subject: [PATCH 2/2] REVERTME: net: stmmac: Fix DMA transfer mode WA in NP
 exit flow

Modify the WA implementation on DMA transfer bit to READ then WRITE
instead of just WRITE to avoid overwrite the BIOS setting on VC/TC.

Signed-off-by: Song, Yoong Siang <yoong.siang.song@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_netproxy.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_netproxy.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_netproxy.c
index eb321120993d..54bf95d261ab 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_netproxy.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_netproxy.c
@@ -134,8 +134,11 @@ irqreturn_t netproxy_irq(int irq, void *dev_id)
 err_skb:
 	/* [REVERTME] DMA_CTL_CH(i) Workaround */
 	for (i = 0; i < EHL_PSE_ETH_DMA_TOTAL_CH; i++) {
-		writel(EHL_PSE_ETH_DMA_MISC_DTM_DRAM, priv->ioaddr
-		       + EHL_PSE_ETH_DMA_MISC_OFFSET + i * sizeof(u32));
+		value = readl(priv->ioaddr + EHL_PSE_ETH_DMA_MISC_OFFSET
+			      + i * sizeof(u32));
+		value |= EHL_PSE_ETH_DMA_MISC_DTM_DRAM;
+		writel(value, priv->ioaddr + EHL_PSE_ETH_DMA_MISC_OFFSET
+		       + i * sizeof(u32));
 	}
 
 	queue_work(priv->netprox_wq, &priv->netprox_task);
-- 
2.20.1

